src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -934,20 +934,20 @@ public unsafe virtual int IndexOf(string source, string value, int startIndex, i
             return IndexOfCore(source, value, startIndex, count, options, null);
         }
 
-        internal int IndexOfOrdinal(ReadOnlySpan<char> source, ReadOnlySpan<char> value, bool ignoreCase)
+        internal int IndexOfOrdinal(ReadOnlySpan<char> source, ReadOnlySpan<char> value, bool ignoreCase, bool fromBeginning = true)
         {
             Debug.Assert(!_invariantMode);
             Debug.Assert(!source.IsEmpty);
             Debug.Assert(!value.IsEmpty);
-            return IndexOfOrdinalCore(source, value, ignoreCase);
+            return IndexOfOrdinalCore(source, value, ignoreCase, fromBeginning);
         }
 
-        internal unsafe int IndexOf(ReadOnlySpan<char> source, ReadOnlySpan<char> value, CompareOptions options)
+        internal unsafe int IndexOf(ReadOnlySpan<char> source, ReadOnlySpan<char> value, CompareOptions options, bool fromBeginning = true)
```

Comment:
>unsafe i [](start = 17, length = 8)

why this is unsafe?

Reply:
if you removed the unsafe, you'll get the compilation error?

---
In reply to: [198905071](https://github.com/dotnet/coreclr/pull/18652#discussion_r198905071) [](ancestors = 198905071)

Reply:
yes